packages/components/src/composite/index.ts

```diff
@@ -1,7 +1,135 @@
-// Originally this pointed at a Reakit implementation of
-// `Composite`, but we are removing Reakit entirely from the
-// codebase. We will continue to support the Reakit API
-// through the 'legacy' version, which uses Ariakit under
-// the hood.
+/**
+ * Composite is a component that may contain navigable items represented by
+ * CompositeItem. It's inspired by the WAI-ARIA Composite Role and implements
+ * all the keyboard navigation mechanisms to ensure that there's only one
+ * tab stop for the whole Composite element. This means that it can behave as
+ * a roving tabindex or aria-activedescendant container.
+ *
+ * @see https://ariakit.org/components/composite
+ */
 
-export * from './legacy';
+/**
+ * External dependencies
+ */
+import {
+	Composite,
+	CompositeGroup,
+	CompositeGroupLabel,
+	CompositeItem,
+	CompositeRow,
+	useCompositeStore,
+} from '@ariakit/react';
+import type { CompositeStore, CompositeStoreProps } from '@ariakit/react';
+
+/**
+ * Creates a composite store.
+ * @see https://ariakit.org/components/composite
+ * @example
+ * ```jsx
+ * const useCompositeStore = Composite.useStore;
+ * const store = useCompositeStore();
+ * <Composite.Root store={store}>
+ *   <Composite.Item>Item</Composite.Item>
+ *   <Composite.Item>Item</Composite.Item>
+ *   <Composite.Item>Item</Composite.Item>
+ * </Composite.Root>
+ * ```
+ */
+export const useStore = useCompositeStore;
+
+/**
+ * Renders a composite widget.
+ * @see https://ariakit.org/components/composite
+ * @example
+ * ```jsx
+ * const useCompositeStore = Composite.useStore;
+ * const store = useCompositeStore();
+ * <Composite.Root store={store}>
+ *   <Composite.Item>Item 1</Composite.Item>
+ *   <Composite.Item>Item 2</Composite.Item>
+ * </Composite.Root>
+ * ```
+ */
+export const Root = Composite;
+
+/**
+ * Renders a group element for composite items.
+ * @see https://ariakit.org/components/composite
+ * @example
+ * ```jsx
+ * const useCompositeStore = Composite.useStore;
+ * const store = useCompositeStore();
```

Comment:
Do we need a store? We could just have a `CompositeProvider` perhaps?

Which makes me wonder, why aren't we re-exporting `CompositeProvider` too?

Reply:
In general, when using ariakit components, we've opted for using the store hooks directly instead of relying on the `Provider` component because of slot fill limitations which would break the provider-based version. By using the store hook explicitly, we can forward the store via a component-dedicated context and get around slot-fill boundaries.

Reply:
Got it, thanks for elaborating!